src/material-experimental/mdc-slider/slider-adapter.ts

```diff
@@ -6,133 +6,179 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import {ChangeDetectorRef, ElementRef, Inject} from '@angular/core';
 import {SpecificEventListener, EventType} from '@material/base';
 import {MDCSliderAdapter, Thumb, TickMark} from '@material/slider';
+import {MatSliderThumb, MAT_SLIDER} from './slider-thumb';
 
+/**
+ * This is a dummy interface that just contains the properties and methods of MatSlider that are
+ * used by SliderAdapter. Rather than directly referencing MatSlider, we use this interface when
+ * to avoid a circular dependency between MatSlider and SliderAdapter.
+ */
+interface MatSlider {
+  _cdr: ChangeDetectorRef;
+  min: number;
+  max: number;
+  disabled: boolean;
+  _elementRef: ElementRef<HTMLElement>;
+  _trackActive: ElementRef<HTMLElement>;
+  _initialized: boolean;
+  _tickMarks: TickMark[];
+  _document: Document;
+  _window: Window;
+  displayWith: ((value: number) => string) | null;
+  _getInput: (thumb: Thumb) => MatSliderThumb;
+  _getKnobElement: (thumb: Thumb) => HTMLElement;
+  _getThumbElement: (thumb: Thumb) => HTMLElement;
+  _getInputElement: (thumb: Thumb) => HTMLInputElement;
+  _setValue: (value: number, thumb: Thumb) => void;
+  _setValueIndicatorText: (value: number, thumb: Thumb) => void;
+}
+
+// TODO(wagnermaciel): Change to prototype methods once this PR is submitted.
+// https://github.com/material-components/material-components-web/pull/6256
 export class SliderAdapter implements MDCSliderAdapter {
-  hasClass(className: string): boolean {
-    throw Error('Method not implemented.');
-  }
-  addClass(className: string): void {
-    throw Error('Method not implemented.');
-  }
-  removeClass(className: string): void {
-    throw Error('Method not implemented.');
-  }
-  getAttribute(attribute: string): string | null {
-    throw Error('Method not implemented.');
-  }
-  addThumbClass(className: string, thumb: Thumb): void {
-    throw Error('Method not implemented.');
-  }
-  removeThumbClass(className: string, thumb: Thumb): void {
-    throw Error('Method not implemented.');
-  }
-  getInputValue(thumb: Thumb): string {
-    throw Error('Method not implemented.');
+  constructor(@Inject(MAT_SLIDER) private readonly _delegate: MatSlider) {}
+  hasClass = (className: string): boolean => {
```

Comment:
The class here should have a JsDoc style description, and then I would move the comment that explains the structure to be in between the constructor and the methods, e.g.
```ts
/** class description here */
export class SliderAdapter  implements MDCSliderAdapter {
  constructor(...) { ... }

  // We manually assign functions instead of using prototype methods because
  // MDC clobbers the values otherwise.
  // See https://github.com/material-components/material-components-web/pull/6256

  hasClass = () => { ... };
}
```